flink-ml-lib/src/main/java/org/apache/flink/ml/common/broadcast/operator/AbstractBroadcastWrapperOperator.java

```diff
@@ -196,17 +203,25 @@ public AbstractBroadcastWrapperOperator(
     }
 
     /**
-     * check whether all of broadcast variables are ready.
+     * checks whether all of broadcast variables are ready. Besides it maintains a state
+     * {broadcastVariablesReady} to avoiding invoking {@code BroadcastContext.isCacheFinished(...)}
+     * repeatedly. Finally, it sets broadcast variables for ${@link HasBroadcastVariable} if the
+     * broadcast variables are ready.
      *
-     * @return
+     * @return true if all broadcast variables are ready, false otherwise.
      */
     protected boolean areBroadcastVariablesReady() {
         if (broadcastVariablesReady) {
             return true;
         }
         for (String name : broadcastStreamNames) {
-            if (!BroadcastContext.isCacheFinished(Tuple2.of(name, indexOfSubtask))) {
+            if (!BroadcastContext.isCacheFinished(name + "-" + indexOfSubtask)) {
                 return false;
+            } else if (wrappedOperator instanceof HasBroadcastVariable) {
+                String key = name + "-" + indexOfSubtask;
+                String userKey = name.substring(name.indexOf('-') + 1);
+                ((HasBroadcastVariable) wrappedOperator)
```

Comment:
Use `OperatorUtils#processOperatorOrUdfIfSatisfy` instead since we may need to handle both of operators and UDF if either of them implements the interface.